documentation/docs/11-ssr-and-javascript.md

```diff
@@ -0,0 +1,77 @@
+---
+title: SSR and JavaScript
+---
+
+By default, SvelteKit will server-render pages on-demand, then **hydrate** the rendered HTML in the client with an interactive Svelte app while initialising a **router** that takes over subsequent navigations.
+
+You can control each of these on a per-app or per-page basis. Note that each of the per-page settings use [`context="module"`](https://svelte.dev/docs#script_context_module), and only apply to page components, _not_ [layout](#layouts) components.
+
+If both are specified, per-page settings override per-app settings in case of conflicts. Each setting can be controlled independently, but `ssr` and `hydrate` cannot both be `false` since that would result in nothing being rendered at all.
+
+### ssr
+
+Disabling server-side rendering effectively turns your SvelteKit app into a **single-page app** or SPA.
+
+> In most situations this is not recommended: it harms SEO, tends to slow down perceived performance, and makes your app inaccessible to users if JavaScript fails or is disabled (which happens [more often than you probably think](https://kryogenix.org/code/browser/everyonehasjs.html)). Sometimes it's appropriate or even necessary, but consider alternatives before disabling SSR.
+
+You can disable SSR app-wide with the [`ssr` config option](#configuration-ssr), or a page-level `ssr` export:
+
+```html
+<script context="module">
+	export const ssr = false;
+</script>
+```
+
+### router
+
+SvelteKit includes a client-side router that intercepts navigations (from the user clicking on links, or interacting with the back/forward buttons) and updates the page contents, rather than letting the browser handle the navigation by reloading.
+
+In certain circumstances you might need to disable this behaviour with the app-wide [`router` config option](#configuration-router) or the page-level `router` export:
```

Comment:
At a page level, what does this do, exactly? Does it use old-fashioned navigation when navigating _from_ this page? To this page? Either?

Reply:
Yeah, the router is disabled while you're on this page. I wasn't sure if we wanted to get into 'disable the router if it's the SSR'd page but enable it if it's already going'

Reply:
I don't really have a strong opinion about how this should work. Having it refer to whether the router is disabled on this page (i.e., when navigating from this page) sounds sensible to me. What I was mostly getting at was that this could probably be made a little more explicit in the docs. (And especially what happens if you use the router to navigate to a page where the router is disabled.)

Reply:
clarified